src/Persisters/Flysystem.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Rubix\ML\Persisters;
+
+use League\Flysystem\FilesystemInterface;
+use Rubix\ML\Encoding;
+use Rubix\ML\Persistable;
+use Rubix\ML\Other\Helpers\Params;
+use Rubix\ML\Persisters\Serializers\Native;
+use Rubix\ML\Persisters\Serializers\Serializer;
+use RuntimeException;
+use Stringable;
+
+/**
+ * Flysystem
+ *
+ * The flysystem persister serializes models to a file at a user-specified path, using a
+ * user-provided flysystem instance.
+ *
+ * Flysystem is a filesystem abstraction providing a unified interface for many different
+ * filesystems. (Local, Amazon S3, Azure Blob Storage, Google Cloud Storage, Dropbox etc...)
+ *
+ * @see https://flysystem.thephpleague.com
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Andrew DalPino
+ */
+class Flysystem implements Persister, Stringable
+{
+    /**
+     * The path to the model file on the filesystem.
+     *
+     * @var string
+     */
+    protected $path;
+
+    /**
+     * The filesystem implementation providing access to your backend storage.
+     *
+     * @var \League\Flysystem\FilesystemInterface
+     */
+    protected $filesystem;
+
+    /**
+     * The serializer used to convert to and from serial format.
+     *
+     * @var \Rubix\ML\Persisters\Serializers\Serializer
+     */
+    protected $serializer;
+
+    /**
+     * @param string $path
+     * @param FilesystemInterface $filesystem
+     * @param \Rubix\ML\Persisters\Serializers\Serializer|null $serializer
+     */
+    public function __construct(string $path, FilesystemInterface $filesystem, ?Serializer $serializer = null)
+    {
+        $this->path = $path;
+        $this->filesystem = $filesystem;
+        $this->serializer = $serializer ?? new Native();
+    }
+
+    /**
+     * Save the persistable model.
+     *
+     * @param \Rubix\ML\Persistable $persistable
+     * @throws \RuntimeException
+     */
+    public function save(Persistable $persistable) : void
+    {
+        $content = $this->serializer->serialize($persistable);
+        $success = $this->filesystem->put($this->path, $content->data());
```

Comment:
Since Encoding objects are 'Stringable' you might not need to called `data()` here ... not sure because I don't know how Flysystem works under the hood but I think you can just pass the Encoding directly as a string type-hinted argument and it will be converted to string under the hood.

Reply:
Yeah, i'll check out what flysystem is doing behind-the-scenes 👍

Reply:
Agreed